sdk/core/core-http/src/serializer.ts

```diff
@@ -1142,36 +1207,88 @@ export interface SimpleMapperType {
     | "any";
 }
 
+/**
+ * Helps build a mapper that describes how to map a set of properties of an object based on other mappers.
+ *
+ * Only one of the following properties should be present: `className`, `modelProperties` and `additionalProperties`.
+ */
 export interface CompositeMapperType {
+  /**
+   * Name of the composite mapper type.
+   */
   name: "Composite";
 
-  // Only one of the two below properties should be present.
-  // Use className to reference another type definition,
-  // and use modelProperties/additionalProperties when the reference to the other type has been resolved.
+  /**
+   * Use className to reference another type definition.
+   */
   className?: string;
 
+  /**
+   * Use `modelProperties` when the reference to the other type has been resolved.
+   */
   modelProperties?: { [propertyName: string]: Mapper };
+
+  /**
+   * Use `additionalProperties` when the reference to the other type has been resolved.
+   */
   additionalProperties?: Mapper;
 
+  /**
+   * The name of the top-most parent scheme, the one that has no parents.
+   */
   uberParent?: string;
+
+  /**
+   * A polymorphic discriminator.
+   */
   polymorphicDiscriminator?: PolymorphicDiscriminator;
 }
 
+/**
+ * Helps build a mapper that describes how to parse a sequence of mapped values.
+ */
 export interface SequenceMapperType {
+  /**
+   * Name of the sequence type mapper.
+   */
   name: "Sequence";
+  /**
+   * The mapper to use to map each one of the properties of the sequence.
+   */
   element: Mapper;
 }
 
+/**
+ * Helps build a mapper that describes how to parse a dictionary of mapped values.
+ */
 export interface DictionaryMapperType {
+  /**
+   * Name of the sequence type mapper.
+   */
   name: "Dictionary";
+  /**
+   * The mapper to use to map the value of each property in the dictionary.
+   */
   value: Mapper;
 }
 
+/**
+ * Helps build a mapper that describes how to parse an enum value.
+ */
 export interface EnumMapperType {
+  /**
+   * Name of the enum type mapper.
+   */
   name: "Enum";
+  /**
+   * Values allowed by this mapper.
+   */
   allowedValues: any[];
 }
 
+/**
+ * The base definition that determines how to map from XML to a JavaScript plain object.
```

Comment:
that base mapper doesn't have to be for XML, it works for vanilla JS too. It just supports both.